File: Coding Exercises/Python/Day50/find_factors.py
import math


def find_factors(num):
    factors = []

    if num <= 0:
        return factors

    limit = int(math.sqrt(num))

    for i in range(1, limit + 1):
        if num % i == 0:
            factors.append(i)
            if i != num // i:
                factors.append(num // i)

    factors.sort()

    return factors

File: Coding Exercises/TypeScript/Day50/findFactors.ts
function findFactors(num: number): number[] {
  let factors: number[] = [];

  if (num <= 0) {
      return factors;
  }

  let limit = Math.floor(Math.sqrt(num));

  for (let i = 1; i <= limit; i++) {
      if (num % i === 0) {
          factors.push(i);
          if (i !== num / i) {
              factors.push(num / i);
          }
      }
  }

  factors.sort((a, b) => a - b);

  return factors;
}
